test-network-function/certification/suite.go

```diff
@@ -173,3 +190,73 @@ func testOperatorCertificationStatus() {
 		}
 	})
 }
+
+func testCSICertified(env *configpkg.TestEnvironment) {
+	testID := identifiers.XformToGinkgoItIdentifier(identifiers.TestCSIOperatorIsCertifiedIdentifier)
+	ginkgo.It(testID, ginkgo.Label(testID), func() {
+		csioperatorsToQuery := env.Csi
+
+		if len(csioperatorsToQuery) == 0 {
+			ginkgo.Skip("No CSI operators to check configured ")
+		}
+
+		ginkgo.By(fmt.Sprintf("Verify operator as certified. Number of CSI drivers to check: %d", len(env.Csi)))
+
+		mapOperatorVersions := csimapping.GetOperatorVersions()
+		ocpVersion := GetOcpVersion()
+		operatorVersionMap, orgMap := GetOperatorVersionMap()
+		testFailed := false
+		for _, csi := range csioperatorsToQuery {
+			pack := csi.Packag
+			org := orgMap[pack]
+			if pack != "" {
+				aKey := csimapping.OperatorKey{OperatorName: pack, OcpVersion: ocpVersion}
+				for _, version := range mapOperatorVersions[aKey] {
+					if operatorVersionMap[pack] == version {
+						tnf.ClaimFilePrintf("Operator: %s ( %s ) currently running version: %s this version is certified to run with Current OCP version %s", org, pack, version, ocpVersion)
+					} else {
+						testFailed = true
+						tnf.ClaimFilePrintf("Operator: %s ( %s ) currently running version: %s this version is NOT certified to run with OCP version %s", org, pack, version, ocpVersion)
+					}
+				}
+			} else {
+				tnf.ClaimFilePrintf("Driver %s is not a certified CSI driver (needs to be part of the operator-certified organization in the catalog) or csimapping.json needs to be updated", csi.Name)
+			}
+		}
+		if testFailed {
+			ginkgo.Fail("At least one CSI operator was not certified to run on this version of openshift. Check Claim.json file for details.")
+		}
+	})
+}
+
+func GetOcpVersion() string {
+	ocCmd := ocpVersionCommand
+	ocVersion := execCommandOutput(ocCmd)
+	nums := strings.Split(ocVersion, ".")
+	ocVersion = strings.Split(nums[0], "\"")[1] + "." + nums[1]
```

Comment:
To improve readability:
**Replace:**
nums := strings.Split(ocVersion, ".")
ocVersion = strings.Split(nums[0], "\"")[1] + "." + nums[1]
**With:**
nums := strings.Split(string.ReplaceAll(ocVersion, "\"","")
ocVersion = strings.Split(nums[0] + "." + nums[1]